tests/OmniSharp.Roslyn.CSharp.Tests/FormattingFacts.cs

```diff
@@ -200,5 +312,51 @@ private static void AssertFormatTargetKind(SyntaxKind kind, string input)
                 }
             }
         }
+
+        private async Task AssertTextChanges(string source, string expected)
+        {
+            var testFile = new TestFile("dummy.cs", source);
+
+            using (var host = CreateOmniSharpHost(testFile))
+            {
+                var span = testFile.Content.GetSpans().Single();
+                var range = testFile.Content.GetRangeFromSpan(span);
+
+                var request = new FormatRangeRequest()
+                {
+                    Buffer = testFile.Content.Code,
+                    FileName = testFile.FileName,
+                    Line = range.Start.Line,
+                    Column = range.Start.Offset,
+                    EndLine = range.End.Line,
+                    EndColumn = range.End.Offset
+                };
+
+                var requestHandler = host.GetRequestHandler<FormatRangeService>(OmniSharpEndpoints.FormatRange);
+
+                var response = await requestHandler.Handle(request);
+                var actual = response.Changes.ToArray();
+
+                var oldText = testFile.Content.Text;
+                var textChanges = GetTextChanges(oldText, response.Changes);
+                var actualText = oldText.WithChanges(textChanges).ToString();
+                Assert.Equal(expected.Replace("\r\n","\n"), actualText.Replace("\r\n","\n"));
+            }
+        }
+
+        public static IEnumerable<TextChange> GetTextChanges(SourceText oldText, IEnumerable<LinePositionSpanTextChange> changes)
+        {
+            var textChanges = new List<TextChange>();
+            foreach( var change in changes)
+            {
+                var startPosition = new LinePosition(change.StartLine, change.StartColumn);
+                var endPosition = new LinePosition(change.EndLine, change.EndColumn);
+                var span = oldText.Lines.GetTextSpan(new LinePositionSpan(startPosition, endPosition));
+                var newText = change.NewText;
+                textChanges.Add(new TextChange(span, newText));
+            }
+
+            return textChanges.OrderBy(change => change.Span);
```

Comment:
I don't think this is necessary.

Reply:
The changes returned by the service are arranged in descending order and the WithChanges function that we are using to get the new text for comparison requires the changes to be arranged in ascending order that is why this line has been added.

Reply:
Have you tested that? I thought SourceText.WithChanges didn't care about the order.

Reply:
Never mind: http://source.roslyn.io/#Microsoft.CodeAnalysis/CodeAnalysisResources.Designer.cs,6a0f623f21709018 :)